src/components/Courses/Courses.jsx

```diff
@@ -32,20 +32,15 @@ function Courses() {
 	};
 
 	const deleteCourse = async (id) => {
-		await makeDeleteRequest(`/courses/${id}`);
-		dispatch(DELETE_COURSES(id));
+		await dispatch(deleteCourses(id));
 	};
 
 	useEffect(() => {
 		async function fetchCourseAndAuthorsList() {
 			try {
-				const response = await makeGetRequest('/authors/all');
-				const res = await response.json();
-
-				dispatch(GET_AUTHORS(res.result));
-				const resp = await makeGetRequest('/courses/all');
-				const result = await resp.json();
-				dispatch(GET_COURSES(result.result));
+				await dispatch(getUserProfile());
+				await dispatch(getAllAuthors());
+				await dispatch(getAllCourses());
```

Comment:
It would be better to fetch the user's info in most parent component (App level), because usually we need user info in all pages except registration and login pages.
Though some websites like ecommerce sites don't need user info in all pages, but there are several pages where we need it and it is recommended to fetch user info once in app level rather fetching in every pages.